services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/FloodlightKafkaConnectionSpec.groovy

```diff
@@ -22,25 +22,35 @@ class FloodlightKafkaConnectionSpec extends BaseSpecification {
         kafkaBreaker.shutoff(KafkaBreakTarget.FLOODLIGHT_PRODUCER)
         kafkaBreaker.shutoff(KafkaBreakTarget.FLOODLIGHT_CONSUMER)
 
-        and: "Remains in this state for 10 seconds"
+        then: "Right before controller alive timeout switches are still active and links are discovered"
+        double interval = 5 * 0.2
+        Wrappers.timedLoop(5 - interval) {
+            assert northbound.activeSwitches.size() == topology.activeSwitches.size()
+            assert northbound.getActiveLinks().size() == topology.islsForActiveSwitches.size() * 2
+            sleep(500)
+        }
+
+        and: "After controller alive timeout switches are inactive and links are discovered"
+        Wrappers.wait(WAIT_OFFSET + interval) { assert northbound.activeSwitches.size() == 0 }
+        northbound.getActiveLinks().size() == topology.islsForActiveSwitches.size() * 2
+
+        when: "System remains in this state for 10 seconds"
```

Comment:
Not sure whether this step still makes sense

Reply:
What makes sense - is to wait for 'discoveryTimeout' time and make sure no links get failed

Reply:
Fixed.